### DISCUSSION TITLE
[Application Development] make devnet-up is throwing  Transaction dropped from the mempool:

### DISCUSSION THREAD

**Discussion by ketanrome** [ethereum-optimism/developers#223] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Running : make devnet-up ( Using tutorials/chain branch)
Sequence of Commands
 rm -rf optimism/
 git clone https://github.com/ethereum-optimism/optimism.git
cd optimism
 git checkout tutorials/chain
 cd optimism/packages/contracts-bedrock
 pnpm install
pnpm build
 cd ../..
make devnet-up

**Environment** 
git          2         2.34.1
go           1.21      1.21.6
node         20        20.11.0
pnpm         8         8.15.0
foundry      0.2.0     0.2.0
make         3         4.3
jq           1.6       1.6
direnv       2         2.25.2

Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy

**Error:** -- Stuck here for sometime. Not sure how to proceed for local environement.
Transaction dropped from the mempool: 0x341cd5ab9577a547d84d8a32196bc3da19509ebeaa08ebd8fbc859ca0cd26b33
Transaction dropped from the mempool: 0xcd2b180640c0252498b99dd50fb3d7175fc1062195e9836efbb168a5a71a51d7
Transaction dropped from the mempool: 0xaa15505e2bc096edc7b087778510a7a435453d87395b2e5f2909bb26217c1951
Transaction dropped from the mempool: 0x6ee14da2f77cedafdf2b56edd48ec00c27a0036dd929b56acacc2aad5b87ab7d
Transaction dropped from the mempool: 0x36e828b3f34da8edad8a02244676bad082896a2cda6ba9a4a64b00c760c33795
Transaction dropped from the mempool: 0x364d4f9d75d631bef01974cee03eaed8f3ca0b5a30ff394fcf847eb7434b16b8
Transaction dropped from the mempool: 0x1333acd0e3aedefa6988241bad4c354423ec0932dab1e242e6d0a5be08240453
Transaction dropped from the mempool: 0xba52534f777d55f51d10ec274b64010c62d5818470b83fc8b8c62824672e17e6
Transaction dropped from the mempool: 0xf18d3cac7fab58262f2fb1a1be8787ae63dfdaae98a2a40907ec00797ec8c900
Transaction dropped from the mempool: 0x8616538d4ba4847e5d81b3a6fc25225aaed5a3c8ca7740bf58ecdc21eea6d051
Transaction dropped from the mempool: 0x0977d37d0ba0d9f9aefca50bacc8911a6cf7c647ee01f0b516a274c995cddedf
Transaction dropped from the mempool: 0x6d60e7e7759f5ef1f619e207edcf3d476bf4eed2c8eb371d1db8a83467c75256
Transaction dropped from the mempool: 0x6302ed6459daba3d6606706d2ff875e8006dc8a952df91870507838ccbb4d8a8
Transaction dropped from the mempool: 0xfc9a6c7ee72a693f5188d29bc8e4fb0d6c765197b4b828e03faec2dd435d3216
Transaction dropped from the mempool: 0x449da8dfbcf394250d33fcd0f338c120cdda1c951f4599dc8f8e216ccb002b4d
Transaction dropped from the mempool: 0x70ea34fe2ab00d80a08f912bc29e7b1f8ea62853ed797782d1aef9ee1a3b7633
Transaction dropped from the mempool: 0xcd2a59b3239313e8c72c294b66950b88a063dcaa37024b12d897d5d6a513d6cd
Transaction dropped from the mempool: 0x6a38da88d8de7e489b6597dfbc0a7be180349e2ab3c21758df1dee2de0882516
Transaction dropped from the mempool: 0x7b9fc12b517c236c84fcaef23970fe174652a8fc50255dfb93af3c29e6183c75
Traceback (most recent call last):
  File "/home/ketan/source/optimism/./bedrock-devnet/main.py", line 9, in <module>
    main()
  File "/home/ketan/source/optimism/./bedrock-devnet/main.py", line 5, in main
    devnet.main()
  File "/home/ketan/source/optimism/bedrock-devnet/devnet/__init__.py", line 98, in main
    devnet_l1_genesis(paths)
  File "/home/ketan/source/optimism/bedrock-devnet/devnet/__init__.py", line 179, in devnet_l1_genesis
    raise Exception(f"Exception occurred in child process: {err}")
Exception: Exception occurred in child process: Command '['forge', 'script', 'scripts/Deploy.s.sol:Deploy', '--sender', '0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266', '--rpc-url', 'http://127.0.0.1:8545', '--broadcast', '--unlocked', '--with-gas-price', '100000000000']' returned non-zero exit status 1.
make[1]: *** [Makefile:155: devnet-allocs] Error 1
make[1]: Leaving directory '/home/ketan/source/optimism'

**Comment by sbvegan** [comment]
Hey @ketanrome, I haven't had a chance to take a deep look at this, but a couple notes really quick:
- the `tutorials/chain` tag is specifically for the [Creating Your Own L2 Rollup Testnet](https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup)
- the devnet is under development to support a future network upgrade. You can see this [draft pr](https://github.com/ethereum-optimism/optimism/pull/9117) for details. The quick context is the local devnet doesn't currently work with 4844 because its using clique mode consensus. When blobs go live, they will be available on the consesnus layer node (i.e. prysm, lighthouse, etc.), so the devnet needs to be updated to reflect the live L1 network.

**Reply by sbvegan** [reply]
Okay I got some more information from Engineering. The devnet runs in our CI pipeline and that is currently working so the latest on develop should work. Can you verify that you have the latest dependencies, specifically can you make sure you have the correct versions of Foundry?

Additional information from an engineer:

> As a part of [this PR](https://github.com/ethereum-optimism/optimism/pull/9292#discussion_r1479166511), I'm going to be adding a check for the installed Foundry version. We rely on some functionality that's been released only as part of nightly builds, so the installed Forge version needs to be fairly recent, but preferably up to date

edit: additional engineering info

**Comment by upnodedev** [comment]
<img width="821" alt="image" src="https://github.com/ethereum-optimism/developers/assets/117493285/d78b6f65-0278-44aa-a0a3-2143c0c39d6e">

We also face this Transaction dropped from the mempool error on Sepolia testnet...

(May relate to not having enough gas in the deployer in this case as Sepolia gas price spike to 60+ gwei)
